java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbstractLocalReader.java

```diff
@@ -215,193 +207,65 @@ private ApduResponse processExplicitAidSelection(SeSelector.AidSelector aidSelec
-   *     the selection process result. When ATR or FCI are not available, they are set to null but
-   *     they can't be both null at the same time.
-   * @throws KeypleReaderIOException if the communication with the reader or the SE has failed
-   */
-  final SelectionStatus openLogicalChannelAndSelect(SeSelector seSelector) {
-
-    SelectionStatus selectionStatus;
-
-    if (seSelector == null) {
-      throw new IllegalArgumentException("Try to open logical channel without selector.");
-    }
-
-    if (!logicalChannelIsOpen) {
-      /*
-       * init of the physical SE channel: if not yet established, opening of a new physical
-       * channel
-       */
-      if (!isPhysicalChannelOpen()) {
-        openPhysicalChannel();
-      }
-      if (!isPhysicalChannelOpen()) {
-        throw new KeypleReaderIOException("Fail to open physical channel.");
-      }
-    }
-
-    selectionStatus = openLogicalChannel(seSelector);
-
-    return selectionStatus;
-  }
-
-  /**
-   * Attempts to open the physical channel
-   *
-   * @throws KeypleReaderIOException if the communication with the reader or the SE has failed
+   * @throws KeypleReaderIOException if the communication with the reader or the SE has failed and
+   *     the physical channel could not be open.
+   * @since 0.9
    */
   protected abstract void openPhysicalChannel();
 
   /**
-   * Closes the current physical channel.
+   * Attempts to close the current physical channel.
    *
-   * <p>This method must be implemented by the ProxyReader plugin (e.g. Pcsc/Nfc/Omapi Reader).
+   * <p>This method must not return normally if the physical channel could not be closed.
    *
    * @throws KeypleReaderIOException if the communication with the reader or the SE has failed
+   * @since 0.9
    */
   protected abstract void closePhysicalChannel();
 
   /**
-   * Tells if the physical channel is open or not
-   *
-   * <p>This method must be implemented by the ProxyReader plugin (e.g. Pcsc/Nfc/Omapi Reader).
+   * Tells if the physical channel is open or not.
    *
-   * @return true is the channel is open
+   * @return True is the physical channel is open, false if not.
+   * @since 0.9
    */
   protected abstract boolean isPhysicalChannelOpen();
 
   /**
-   * Tells if a logical channel is open
+   * (package-private)<br>
+   * Tells if a logical channel is open or not.
    *
-   * @return true if the logical channel is open
+   * @return True if the logical channel is open, false if not.
+   * @since 0.9
    */
   final boolean isLogicalChannelOpen() {
     return logicalChannelIsOpen;
   }
 
-  /** Close the logical channel. */
+  /**
+   * Close the logical channel.<br>
+   *
+   * @since 0.9
+   */
   private void closeLogicalChannel() {
     if (logger.isTraceEnabled()) {
       logger.trace("[{}] closeLogicalChannel => Closing of the logical channel.", this.getName());
     }
     logicalChannelIsOpen = false;
```

Comment:
in addition, in case of SmartSelectionReader => need to call SmartSelectionReader.closeLogicalChannel()

Reply:
added